lib/ansible/modules/cloud/vmware/vmware_object_role_permission.py

```diff
@@ -0,0 +1,230 @@
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: vmware_object_role_permission
+short_description: This module can be used to manage object permissions on the given host.
+description:
+    - Manage local roles on an ESXi host
+version_added: "2.6"
+author:
+- Derek Rushing (@kryptsi) <drush@kryptsi.com>
+notes:
+    - Tested on ESXi 6.5
+    - Be sure that the ESXi user used for login, has the appropriate rights to administer permissions
+requirements:
+    - "python >= 2.7"
+    - PyVmomi
+options:
+    role:
+        description:
+            - The role to be assigned permission.
+        required: True
+    principal:
+        description:
+            - The user to be assigned permission.  Required if group is not specified.
+    group:
+        description:
+            - The group to be assigned permission. Required if principal is not specified.
+    object_name:
+        description:
+            - The object name to assigned permission.
+        required: True
+    object_type:
+        description:
+            - The object type being targeted.
+        default: 'Folder'
+        choices: ['Folder', 'VirtualMachine', 'Datacenter', 'ResourcePool', 'Datastore', 'Network', 'HostSystem',
+                  'cluster', 'ClusterComputeResource', 'DistributedVirtualSwitch']
+    recursive:
+        description:
+            - Should the permissions be recursively applied.
+        default: True
+        type: bool
+    state:
+        description:
+            - Indicate desired state of the object's permission. When C(state=present), the permission will be added if it doesn't already exist.
+            - When C(state=absent), the permission is removed if it exists.
+        choices: ['present', 'absent']
+        default: present
+extends_documentation_fragment: vmware.documentation
+'''
+
+EXAMPLES = '''
+# Example vmware_object_permission command from Ansible Playbooks
+- name: Assign user to VM folder
+  vmware_object_role_permission:
+      role: administrator
+      principal: user_bob
+      object_name: services
+      state: present
+
+- name: Remove user from VM folder
+  vmware_object_role_permission:
+      role: administrator
+      principal: user_bob
+      object_name: services
+      state: absent
+
+- name: Assign finance group to VM folder
+  vmware_object_role_permission:
+      role: Limited Users
+      group: finance
+      object_name: Accounts
+      state: present
+
+'''
+
+RETURN = r'''
+changed:
+    description: whether or not a change was made to the object's role
+    returned: always
+    type: bool
+'''
+
+try:
+    from pyVmomi import vim, vmodl
+except ImportError:
+    pass
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.vmware import (PyVmomi, vmware_argument_spec,
+                                         find_obj, connect_to_api,
```

Comment:
connect_to_api and wait_for_task is not being used.

Reply:
Removed